apps/performance-tests/src/hierarchies/Filtering.test.ts

```diff
@@ -0,0 +1,107 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { IModelDb, PhysicalElement, SnapshotDb } from "@itwin/core-backend";
+import { createNodesQueryClauseFactory, HierarchyFilteringPath } from "@itwin/presentation-hierarchies";
+import { Datasets } from "../util/Datasets";
+import { run } from "../util/TestUtilities";
+import { ProviderOptions, StatelessHierarchyProvider } from "./StatelessHierarchyProvider";
+import { createBisInstanceLabelSelectClauseFactory, ECClassHierarchyInspector, ECSchemaProvider } from "@itwin/presentation-shared";
+
+describe("filtering", () => {
+  const totalNumberOfFilteringPaths = 50000;
+  const physicalElementsSmallestIndex = 20;
+
+  run({
+    testName: `filters with ${totalNumberOfFilteringPaths} paths`,
+    setup: (): ProviderOptions => {
+      const { schemaName, itemsPerGroup, defaultClassName } = Datasets.CUSTOM_SCHEMA;
+
+      const filtering = {
+        paths: new Array<HierarchyFilteringPath>(),
+      };
+      const parentIdsArr = new Array<number>();
+      for (let i = 1; i <= 100; ++i) {
+        parentIdsArr.push(i + physicalElementsSmallestIndex);
+        for (let j = (i - 1) * 500; j < i * 500; ++j) {
+          filtering.paths.push([
+            { className: `${schemaName}.${defaultClassName}_0`, id: `0x${physicalElementsSmallestIndex.toString(16)}` },
+            { className: `${schemaName}.${defaultClassName}_${i / itemsPerGroup}`, id: `0x${(i + physicalElementsSmallestIndex).toString(16)}` },
+            { className: `${schemaName}.${defaultClassName}_${j / itemsPerGroup}`, id: `0x${(j + physicalElementsSmallestIndex).toString(16)}` },
+          ]);
+        }
+      }
+
+      const iModel = SnapshotDb.openFile(Datasets.getIModelPath("50k flat elements"));
+      const fullClassName = PhysicalElement.classFullName.replace(":", ".");
+      const createHierarchyLevelDefinition = async (imodelAccess: ECSchemaProvider & ECClassHierarchyInspector, eCInstanceIdCondition: string) => {
```

Comment:
- letter casing problem in `eCInstanceIdCondition`
- I suggest changing it to `whereClause: (alias: string) => string`

Reply:
Changed